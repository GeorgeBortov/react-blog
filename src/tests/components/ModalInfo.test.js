import React from 'react';
import { shallow } from 'enzyme';
import ModalInfo from '../../components/ModalInfo';

let onRequestClose;

beforeEach(() => {
    onRequestClose = jest.fn();
    
});

test('should render close ModalInfo correctly', () => {
    const wrapper = shallow(<ModalInfo
        isOpen={false}
        onRequestClose={onRequestClose}
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should render open ConfirmModal correctly', () => {
    const wrapper = shallow(<ModalInfo
        isOpen={true}
        onRequestClose={onRequestClose}
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should handle onRequestClose with button "Ok" ', () => {
    const wrapper = shallow(<ModalInfo
        onRequestClose={onRequestClose}
    />);
    wrapper.find('.removeExpense').simulate('click');
    expect(onRequestClose).toHaveBeenCalled();
});

test('should handle onRequestClose with button "X" ', () => {
    const wrapper = shallow(<ModalInfo
        onRequestClose={onRequestClose}
    />);
    wrapper.find('.modal__close').simulate('click');
    expect(onRequestClose).toHaveBeenCalled();
});