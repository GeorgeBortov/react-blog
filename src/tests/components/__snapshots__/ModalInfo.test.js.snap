// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render close ModalInfo correctly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation Modal"
  isOpen={false}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__body"
  >
    <button
      className="modal__close"
      onClick={[MockFunction]}
    />
    <h3
      className="modal__title"
    >
      Please check Your email. We send you a letter.
    </h3>
    <button
      className="button removeExpense"
      onClick={[MockFunction]}
    >
      Ok
    </button>
  </div>
</Modal>
`;

exports[`should render open ConfirmModal correctly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation Modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__body"
  >
    <button
      className="modal__close"
      onClick={[MockFunction]}
    />
    <h3
      className="modal__title"
    >
      Please check Your email. We send you a letter.
    </h3>
    <button
      className="button removeExpense"
      onClick={[MockFunction]}
    >
      Ok
    </button>
  </div>
</Modal>
`;
